src/logic/calculate.js

```diff
@@ -0,0 +1,22 @@
+import operate from './operate';
+
+const calculate = (data, btnName) => {
+  const temp = data;
+  const calcOperations = {
```

Comment:
i dont know why you need 
```
const temp = data;
```
`data` is an object, that will copy by reference, so basically 
 `temp.total` and  `data.total` will be the same thing

Reply:
Also you need to destructure `data` here (something like) :
```
 let { total, next, operation } = data;
``` 
so that you get to be using `total` simply, instead of `data.total`

Reply:
I am not exactly sure if destructuring (copying into new variables) will help me directly modify the data object, as stated in the milestone.

I made a shallow copy so I can mutate the data object directly. I am open to better ways to achieve this.

Reply:
You can manipulate individual variables, and if you need the object at the end, then you can get  it back by making the object 🙂
If I want `var1` and `var2` as object , I just make the object `{var1, var2}` 

Just to say it's doable 🙂

Reply:
Aha !!! :smile: I see your perspective. Didn't think about that earlier. 
Thanks for the hint @shloch